actions/views/rhs.jsx

```diff
@@ -0,0 +1,176 @@
+
+export function updateCommentDraft(rootId, draft) {
+    return (dispatch) => {
+        PostStore.storeCommentDraft(rootId, draft);
+
+        dispatch({
+            type: ActionTypes.POST_DRAFT_UPDATED,
+            data: {rootId, draft}
+        });
+    };
+}
+
+export function makeOnMoveHistoryIndex(rootId, direction) {
+    const getMessageInHistory = makeGetMessageInHistoryItem(Posts.MESSAGE_TYPES.COMMENT);
+
+    return () => (dispatch, getState) => {
+        const draft = PostStore.getCommentDraft(rootId);
+
+        if (draft.message !== '' && draft.message !== getMessageInHistory(getState())) {
+            return;
+        }
+
+        if (direction === -1) {
+            dispatch(moveHistoryIndexBack(Posts.MESSAGE_TYPES.COMMENT));
+        } else if (direction === 1) {
+            dispatch(moveHistoryIndexForward(Posts.MESSAGE_TYPES.COMMENT));
+        }
+
+        const nextMessageInHistory = getMessageInHistory(getState());
+
+        dispatch(updateCommentDraft(rootId, {...draft, message: nextMessageInHistory}));
+    };
+}
+
+export function submitPost(channelId, rootId, draft) {
+    return (dispatch, getState) => {
+        const state = getState();
+
+        const userId = getCurrentUserId(state);
+
+        const time = Utils.getTimestamp();
+
+        const post = {
+            file_ids: [],
+            message: draft.message,
+            channel_id: channelId,
+            root_id: rootId,
+            parent_id: rootId,
+            pending_post_id: `${userId}:${time}`,
+            user_id: userId,
+            create_at: time
+        };
+
+        GlobalActions.emitUserCommentedEvent(post);
+
+        PostActions.createPost(post, draft.fileInfos);
+    };
+}
+
+export function submitReaction(postId, action, emojiName) {
+    return (dispatch) => {
+        if (action === '+') {
+            dispatch(addReaction(postId, emojiName));
+        } else if (action === '-') {
+            dispatch(removeReaction(postId, emojiName));
+        }
+    };
+}
+
+export function submitCommand(channelId, rootId, draft) {
+    return (dispatch, getState) => {
+        const state = getState();
+
+        const teamId = getCurrentTeamId(state);
+
+        const args = {
+            channel_id: channelId,
+            team_id: teamId,
+            root_id: rootId,
+            parent_id: rootId
+        };
+
+        const {message} = draft;
+
+        return new Promise((resolve, reject) => {
+            ChannelActions.executeCommand(message, args, resolve, (err) => {
+                if (err.sendMessage) {
+                    dispatch(submitPost(channelId, rootId, draft));
+                } else {
+                    reject(err);
+                }
+            });
+        });
+    };
+}
+
+export function makeOnSubmit(channelId, rootId, latestPostId) {
+    return () => async (dispatch, getState) => {
+        const draft = PostStore.getCommentDraft(rootId);
```

Comment:
Similar to comment above, maybe use a storage selector to get the draft? Or create a new selector for it. I'm 0/5 on how it's done, just want to reduce PostStore usage

Reply:
Ah yeah I see there was a push recently with all this in it. I'll make use of it 👍

Reply:
@jwilander removed PostStore usage